### PR TITLE
Egg Game unit test improvements

### PR DESCRIPTION
**Pull request by kswenson**
Add unit tests for
- new GeneticsUtils functions
  - alleleStringContainsAlleles()
  - isValidAlleleString()
  - isCompleteAlleleString()
- TrialGenerator
  - all-combinations
  - randomize-order
- Redux actions
  - changeBasketSelection()
  - changeDrakeSelection()
  - acceptEggInBasket()
  - rejectEggFromBasket()
  - navigateToCurrentRoute()
  - submitEggForBasket()

Sam got the last one, so this one's for @dstrawberrygirl.

### CODE REVIEW

File: test/actions/egg-rejected.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    expect(dispatch.calls.length).toBe(1);
+  });
+
+  it('should create an action to reject the egg from the basket and show completion dialog', () => {
+    const eggDrakeIndex = 0, basketIndex = 0, isChallengeComplete = true,
+          rejectEggArgs = { eggDrakeIndex, basketIndex, isChallengeComplete },
+          rejectEggAction = { type: types.EGG_REJECTED, eggDrakeIndex, basketIndex },
+          showCompleteChallengeAction = {
+                                          type: types.MODAL_DIALOG_SHOWN,
+                                          message: "~ALERT.TITLE.GOOD_WORK",
```
Comment (dstrawberrygirl) [inline test/actions/egg-rejected.js]: Is this the correct message to show for a rejection?

Comment (kswenson) [inline test/actions/egg-rejected.js]: I agree that it seems a little weird, but Egg Game III is different than other challenges in that you're done when you finish the last egg, independently of whether you get it correct or not. First there's a SUBMIT_EGG action which triggers a correct/incorrect message. Then when you dismiss the initial feedback dialog, an EGG_ACCEPTED/EGG_REJECTED action is dispatched, either of which can result in finishing the challenge. So the "Good work!" message is for completing the challenge, not necessarily for sorting the last egg properly (for which feedback has already been given).

File: test/utilities/genetics-utils.js [hunk 2]
```diff
@@ -69,6 +69,67 @@ describe("GeneticsUtils.convertDashAllelesObjectToABAlleles()", function() {
   });
 });
 
+describe("GeneticsUtils.alleleStringContainsAlleles()", function() {
+
```
Comment (dstrawberrygirl) [inline test/utilities/genetics-utils.js]: I'm not seeing a case for if a null is passed in here - the other tests here do cover that.

Comment (kswenson) [inline test/utilities/genetics-utils.js]: I'll add it.

**Pull request merged**